### PR TITLE
feat: GA Gax HttpJson

### PR DESCRIPTION
**Pull request by lqiu96**
We will begin the process of trying to GA Gax's HttpJson module. `@BetaApi` will slowly be removed where it makes sense as we have observed stability in this module.

### CODE REVIEW

File: gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java [hunk 1]
```diff
@@ -326,7 +326,6 @@ public Duration getStreamIdleTimeout() {
     return streamIdleTimeout;
   }
 
-  @BetaApi("The surface for extra headers is not stable yet and may change in the future.")
```
Comment (lqiu96) [inline gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java:329]: Issue to better support extraHeaders: https://github.com/googleapis/sdk-platform-java/issues/1752

Will be done in the future.

Comment (lqiu96) [reply gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java:329]: If this BetaApi annotation is removed in gax-httpjson, should it also be removed in gax and gax-grpc?

**Comment by lqiu96** [comment]
@blakeli0 @suztomo @ddixit14 I'm looking to begin the process of GAing Gax-HttpJson. I want to see your opinions of what version I should set for this module's GA: v1.0.0 or v2.30.0 (should match the next version of gax & gax-grpc).

Immediately bumping to v2.30.0 would keep all the gax-module versions in sync, but may be confusing to customers who see the bump from v0.114.1 -> v2.30.0.

File: gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java (hunk 1 above)
Comment (blakeli0) [reply gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java:329]: If the base class/interface is still `BetaApi`, we don't have to remove them, I would only remove `BetaApi` for httpjson specific classes.

Comment (lqiu96) [reply gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java:329]: > If the base class/interface is still `BetaApi`, we don't have to remove them, I would only remove `BetaApi` for httpjson specific classes.

Just to clarify what you mean by httpjson specific classes. Do you mean:
1. Remove annotation for classes/ methods that exist ONLY in the httpjson module and no other gax module (`withExtraHeaders` would keep the BetaApi annotation)
2. Remove annotation only from the httpjson module (i.e. Remove in httpjson but keep in gax and gax-grpc)

Comment (blakeli0) [reply gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java:329]: 1 yes, 2 no. In addition, if a class does not extend from classes in other modules, we should remove `BetaApi` from it unless there is a reason not to(e.g. maybe the feature in grpc is also still in Beta).

**Comment by suztomo** [comment]
I'm for "v2.30.0 (should match the next version of gax & gax-grpc)". The consistent version gives clarity in error messages when a dependency conflict happens.

File: gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java (hunk 1 above)
Comment (lqiu96) [reply gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java:329]: Gotcha. I'll check the files again and update.

**Comment by lqiu96** [comment]
Things left with BetaApi annotation:
1. HttpJsonCallContext: `withExtraHeaders` and `getExtraHeaders` - HttpJsonCallContext extends from ApiCallContext and the methods override from there. The parent methods still have BetaApi annotation

Note:
1. HttpJsonOperationSnapshotCallable extends from UnaryCallable, but UnaryCallable has no BetaApi annotation (also oddly enough has both BetaApi and Internal annotations)
2. HttpJsonStatusCode extends from StatusCode, but StatusCode has no BetaApi annotation

Weird behaviors:
1. InstantiatingHttpJsonChannelProvider extends from TransportChannelProvider. TransportChannelProvider has two BetaApi annotations in `needsCredentials` and `withCredentials`, but InstantiatingHttpJsonChannelProvider's overriden methods don't have BetaApi annotations

**Comment by lqiu96** [comment]
> I'm for "v2.30.0 (should match the next version of gax & gax-grpc)". The consistent version gives clarity in error messages when a dependency conflict happens.

I've updated the current for gax-httpjson to be `2.29.1-SNAPSHOT` and I believe with this `feat:` PR, it'll mark it as `2.30.0`. If anyone has any concerns, let me know.

**Comment by lqiu96** [comment]
@suztomo @blakeli0 @ddixit14 Would you all be able to review this PR? If there are any concerns, I can target a different cycle to unblock the release train.

File: coverage-report/pom.xml [hunk 2]
```diff
@@ -41,7 +41,7 @@
             <dependency>
                 <groupId>com.google.api</groupId>
                 <artifactId>gax-httpjson</artifactId>
-                <version>0.114.1-SNAPSHOT</version> <!-- {x-version-update:gax-httpjson:current} -->
+                <version>2.29.1-SNAPSHOT</version> <!-- {x-version-update:gax-httpjson:current} -->
```
Comment (suztomo) [inline coverage-report/pom.xml:44]: Can you remove `{x-version-update:gax-httpjson:current}` altogether in favor of `version.gax`

**Comment by suztomo** [comment]
HttpJsonCallContext's withExtraHeaders is marked as BetaApi. Is this intentional (I'm not saying you should remove it).

```
~/sdk-platform-java/gax-java/gax-httpjson $ grep -ilr -A 2 '@BetaApi' .                    git[branch:main-gax_httpjson_GA]
./src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java
suztomo@suztomo-macbookpro.roam.internal ~/sdk-platform-java/gax-java/gax-httpjson
~/sdk-platform-java/gax-java/gax-httpjson $ grep -ir -A 2 '@BetaApi' .                     git[branch:main-gax_httpjson_GA]
./src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java:  @BetaApi("The surface for extra headers is not stable yet and may change in the future.")
./src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java-  @Override
./src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java-  public ApiCallContext withExtraHeaders(Map<String, List<String>> extraHeaders) {
--
./src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java:  @BetaApi("The surface for extra headers is not stable yet and may change in the future.")
./src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java-  @Override
./src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java-  public Map<String, List<String>> getExtraHeaders() {

```

File: coverage-report/pom.xml (hunk 2 above)
Comment (lqiu96) [reply coverage-report/pom.xml]: Ah, good idea! Updated in this file as well as the gax-bom. I noticed that we still have `gax-grpc`. Shall I remove this that reference as well (or maybe in a different PR)?

**Review by suztomo — APPROVED** [review]

**Pull request merged**